pkg/controller/controller_utils.go

```diff
@@ -420,3 +421,21 @@ func FilterActivePods(pods []api.Pod) []*api.Pod {
 	}
 	return result
 }
+
+func SyncAllPodsWithStore(kubeClient client.Interface, store cache.Store) {
+	var allPods *api.PodList
+	var err error
+	listOptions := api.ListOptions{LabelSelector: labels.Everything(), FieldSelector: fields.Everything()}
+	for {
+		if allPods, err = kubeClient.Pods(api.NamespaceAll).List(listOptions); err != nil {
```

Comment:
Isn't it better to just wait for the store to get synced? There's a member function on store you can call to see if it's gotten a complete sync or not.